google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/QueryOptionsTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner.connection;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Statement;
+import com.google.spanner.v1.ExecuteSqlRequest;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class QueryOptionsTest extends AbstractMockServerTest {
+
+  @Test
+  public void testUseOptimizerVersionFromConnectionUrl() {
+    try (Connection connection = createConnection("?optimizerVersion=10")) {
+      Repeat.twice(
+          () -> {
+            executeSelect1AndConsumeResults(connection);
+
+            assertEquals(1, mockSpanner.countRequestsOfType(ExecuteSqlRequest.class));
+            ExecuteSqlRequest request =
+                mockSpanner.getRequestsOfType(ExecuteSqlRequest.class).get(0);
+            assertEquals("10", request.getQueryOptions().getOptimizerVersion());
+
+            mockSpanner.clearRequests();
+          });
+    }
+  }
+
+  @Test
+  public void testUseOptimizerVersionFromStatement() {
+    try (Connection connection = createConnection()) {
```

Comment:
Would it make sense to also pass optimizerVersion here and verify that its overridden?

Reply:
It's a good point. But this test is not intended to specifically verify that, and we also have specific tests that cover that here: https://github.com/googleapis/java-spanner/blob/aa9ad7f5a5e2405e8082a542916c3d1fa7d0fa25/google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionTest.java#L60